### PR TITLE
fix(VDataTableHeader): decrease sort icon line height

### LINKED ISSUE #12823: [Bug Report] Fixed table header border broken in dialog
### Environment
**Vuetify Version:** 2.3.21
**Vue Version:** 2.6.12
**Browsers:** Chrome 87.0.4280.88
**OS:** Mac OS 11.1.0

### Steps to reproduce
I've found some weird issue when having a fixed table in a dialog. At certain heights, the table header line is broken and only shows a line over the checkbox. If you change the height to `height="calc(100vh - 201px)"`, the issue seems to be fixed? 

### Expected Behavior
A consistent table separator when in a dialog. 

### Actual Behavior
The line breaks.
Screen Shot 2020-12-22 at 3 09 46 PM

### Reproduction Link
<a href="https://codepen.io/joebayld/pen/PoGJGvq" target="_blank">https://codepen.io/joebayld/pen/PoGJGvq</a>

### PR DESCRIPTION
**Pull request by yuwu9145**
fixes #12823

## Description

fixes #12823

**Cause**: when table header column has a sorting arrow icon, icon eventual height is bigger than its column header text `span` and will have impact on parent `th` `box-shadow` 

**Solution**: in `VDataTableHeader` level, set icon's `line-height: 0.9` to make sure its eventual shape does not exceeds header column bottom line.

## Motivation and Context

close #12823

## How Has This Been Tested?

visually

## Markup:

<details>

```vue
<template>
  <v-container>
    <v-dialog :value="true">
      <v-card>
        <v-data-table
          dark
          dense
          show-select
          single-select
          fixed-header
          height="calc(100vh - 200px)"
          :items-per-page="15"
          :headers="headers"
          :items="items"
        >
        </v-data-table>
       </v-card>
    </v-dialog>
  </v-container>
</template>

<script>
  export default {
    data () {
      return {
        headers: [
          { text: "Name", value: "name" },
          { text: "Manufacture", value: "manufacture" }
        ],
        items: [],
      }
    },
    created() {
      for (let i =0; i <= 1000; i++) {
        this.items.push({
            id: i,
            name: "Test",
            manufacture: "Test",
          })
      }
      console.log('created')
    }
  }
</script>
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VDataTable/VDataTableHeader.sass [hunk 1]
```diff
@@ -27,6 +27,8 @@
       pointer-events: auto
       cursor: pointer
       outline: 0
+      .v-data-table-header__icon
+        line-height: .9
```
Comment (johnleider) [inline packages/vuetify/src/components/VDataTable/VDataTableHeader.sass:31]: What was the determining factor of this value?

Comment (yuwu9145) [reply packages/vuetify/src/components/VDataTable/VDataTableHeader.sass:31]: Hi @johnleider ,

Thanks for the comment.
There is no specific reason for this value. 

However, by assigning line height to `0.9`, height of this dom will be decreased down to `20px` which ensures its size does not exceed header column bottom line.

This is only solution so far I can think of by touching neither `font-size` nor anything that affects the icon's eventual size.

Please advice any better solution.  

2021_01_06_0kc_Kleki

Comment (johnleider) [reply packages/vuetify/src/components/VDataTable/VDataTableHeader.sass:31]: Thoughts @nekosaur ?

**Comment by yuwu9145** [comment]
Hi @nekosaur , would you mind checking this out if you got a time? It is just a simple css change. :)

**Review by nekosaur — APPROVED** [review]

**Pull request merged**